robotbase/src/main/java/com/team2357/frc2022/subsystems/SensorSubsystem.java

```diff
@@ -32,27 +45,60 @@ public SensorSubsystem(SensorBooleanState intakeSensor, SensorBooleanState feede
         m_currentCargoCount = 0;
         m_cargoAcquired = 0;
         m_cargoLaunched = 0;
+
+        m_intakeDebounceCounter = 0;
+        m_feederDebounceCounter = 0;
+
+        m_nextReadMillis = 0;
+    }
+
+    public void configure(Configuration config) {
+        m_config = config;
     }
 
     @Override
     public void periodic() {
-        boolean intakeState = m_intakeSensor.getState();
-        boolean feederState = m_feederSensor.getState();
+        
+        if (System.currentTimeMillis() > m_nextReadMillis) {
+            boolean intakeState = m_intakeSensor.getState();
+            boolean feederState = m_feederSensor.getState();
+
 
-        if (intakeState != m_lastIntakeState) {
-            if (intakeState) {
-                m_currentCargoCount++;
-                m_cargoAcquired++;
+            if (intakeState == m_lastIntakeState && m_intakeDebounceCounter > 0) {
+                m_intakeDebounceCounter--;
+            } else if (intakeState != m_lastIntakeState) {
```

Comment:
I'm having a hard time understanding exactly what this is doing. Let's talk about it in person